### DISCUSSION TITLE
How to install pytest from source code?

### DISCUSSION THREAD

**Discussion by yongjames** [pytest-dev/pytest#9665] [General] [answered]
‘pip install pytest’ cannot be used because there is no Internet connection on the device,
and there is no source code installation guide in README, so how to install pytest from source code?

**Comment by The-Compiler** [comment, marked as answer]
Like any other Python project, something like `pip install .`. See the [packaging.python.org docs](https://packaging.python.org/en/latest/tutorials/installing-packages/#installing-from-a-local-src-tree).

However, that won't change much, as you still need pytest's dependencies. You might want to use `pip download pytest` and install from the resulting wheels, or perhaps use [devpi](https://www.devpi.net/) as a caching proxy.

**Reply by yongjames** [reply]
much thanks, it's work

**Comment by nicoddemus** [comment]
> However, that won't change much, as you still need pytest's dependencies.

Just a small correction, `pip install .` also installs the dependencies. 👍
